Feature/orders

Shamskol opened the pull request.

Shamskol committed:
feat: add orders to database

Shamskol committed:
created the orderModel and orderRouter and the order migrations table

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,297 @@
+ *          type: string
+ *        avatarUrl:
+ *          type: string
+ *          description: public url of profile avatar
+ *      example:
+ *        id: '00uhjfrwdWAQvD8JV4x6'
+ *        email: 'frank@example.com'
+ *        name: 'Frank Martinez'
+ *        avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/hermanobrother/128.jpg'
+ *
+ * /profiles:
+ *  get:
+ *    description: Returns a list of profiles
+ *    summary: Get a list of all profiles
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - profile
+ *    responses:
+ *      200:
+ *        description: array of profiles
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: array
+ *              items:
+ *                $ref: '#/components/schemas/Profile'
+ *              example:
+ *                - id: '00uhjfrwdWAQvD8JV4x6'
+ *                  email: 'frank@example.com'
+ *                  name: 'Frank Martinez'
+ *                  avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/hermanobrother/128.jpg'
+ *                - id: '013e4ab94d96542e791f'
+ *                  email: 'cathy@example.com'
+ *                  name: 'Cathy Warmund'
+ *                  avatarUrl: 'https://s3.amazonaws.com/uifaces/faces/twitter/geneseleznev/128.jpg'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      403:
+ *        $ref: '#/components/responses/UnauthorizedError'
```
This needs to be updated for orders

Shamskol replied:
Updated

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,297 @@
+
+/**
+ * @swagger
+ * components:
+ *  parameters:
+ *    profileId:
+ *      name: id
+ *      in: path
+ *      description: ID of the profile to return
+ *      required: true
+ *      example: 00uhjfrwdWAQvD8JV4x6
+ *      schema:
+ *        type: string
+ *
+ * /profile/{id}:
+ *  get:
+ *    description: Find profiles by ID
+ *    summary: Returns a single profile
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - profile
+ *    parameters:
+ *      - $ref: '#/components/parameters/profileId'
+ *    responses:
+ *      200:
+ *        description: A profile object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              $ref: '#/components/schemas/Profile'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        description: 'Profile not found'
```
Same with this

Shamskol replied:
Updated

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,297 @@
+/**
+ * @swagger
+ * /profile:
+ *  post:
+ *    summary: Add a profile
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - profile
+ *    requestBody:
+ *      description: Profile object to to be added
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Profile'
+ *    responses:
+ *      400:
+ *        $ref: '#/components/responses/BadRequest'
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        description: 'Profile not found'
+ *      200:
+ *        description: A profile object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: profile created
+ *                profile:
+ *                  $ref: '#/components/schemas/Profile'
```
This one too

Shamskol replied:
Updated

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,297 @@
+});
+/**
+ * @swagger
+ * /profile:
+ *  put:
+ *    summary: Update a profile
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - profile
+ *    requestBody:
+ *      description: Profile object to to be updated
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Profile'
+ *    responses:
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        $ref: '#/components/responses/NotFound'
+ *      200:
+ *        description: A profile object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: profile created
+ *                profile:
+ *                  $ref: '#/components/schemas/Profile'
```
And this one

Shamskol replied:
Updated

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,297 @@
+  } else {
+    res.status(404).json({ message: 'Order missing' });
+  }
+});
+/**
```
We are missing a new line here

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,297 @@
+  }
+});
+/**
+ * @swagger
+ * /profile/{id}:
+ *  delete:
+ *    summary: Remove a profile
+ *    security:
+ *      - okta: []
+ *    tags:
+ *      - profile
+ *    parameters:
+ *      - $ref: '#/components/parameters/profileId'
+ *    responses:
+ *      401:
+ *        $ref: '#/components/responses/UnauthorizedError'
+ *      404:
+ *        $ref: '#/components/responses/NotFound'
+ *      200:
+ *        description: A profile object
+ *        content:
+ *          application/json:
+ *            schema:
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: Profile '00uhjfrwdWAQvD8JV4x6' was deleted.
+ *                profile:
+ *                  $ref: '#/components/schemas/Profile'
```
This one too

Shamskol replied:
Updated

wSedlacek commented on data/migrations/20201006184222_orders.js:
```diff
@@ -0,0 +1,29 @@
+      table.string('dateOrdered').notNullable();
+      table.string('status').notNullable();
+      table.integer('price');
+      table.boolean('priceDetermined').notNullable();//???
```
Yes this is not nullable, we can remove the comment.

wSedlacek commented on api/profile/profileRouter.js:
```diff
@@ -167,7 +167,7 @@ router.post('/', authRequired, async (req, res) => {
         if (pf == undefined) {
           //profile not found so lets insert it
-          await Profiles.create(profile).then((profile) =>
+          await Profiles.create(profile).then((profile) =>//naming convention duplicated
```
We should remove this comment as there is no reason to change this file in this PR

Shamskol replied:
Comment removed

Shamskol committed:
npm run lint:fix to address the formatting issue

Shamskol committed:
Dealt with all the review comments

Shamskol committed:
Handled formatting issues with npm run lint:fix

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,102 @@
+router.post('/', authRequired, async (req, res) => {
+  const newOrder = req.body;
+  if (newOrder) {
+    const id = newOrder.id || 0; //0 needs to be replaced with a string
```
0 needs to be replaced here before merging.
Can you look into `cuid` and the `uuid` libraries and use either one of those to replace this `0`?

wSedlacek commented on api/order/orderRouter.js:
```diff
@@ -0,0 +1,102 @@
+router.put('/', authRequired, function (req, res) {
+  const order = req.body;
+  if (order) {
+    const id = order.id || 0; // 0 needs to be replaced
```
Here too

Commit:
Merge remote-tracking branch 'origin/main' into feature/orders

Commit:
refactor: clean up types

This uses JS Doc TypeScript annotations to clean up typing and provides auto complete

Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>

Commit:
perf: filter list in DB rather than in JavaScript

Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>

Commit:
fix: drep in filter update

Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>

shighetari committed:
test

wSedlacek commented on __tests__/routes/orders.test.js:
```diff
@@ -0,0 +1,86 @@
+
+
+
+//   describe('POST /order', () => {
+//     it('should return 200 when order is created', async () => {
+//       const profile = {
+//         name: 'Louie Smith',
+//         email: 'louie@example.com',
+//         avatarUrl:
+//           'https://s3.amazonaws.com/uifaces/faces/twitter/hermanobrother/128.jpg',
+//       };
+//       Profil.findById.mockResolvedValue(undefined);
+//       Profiles.create.mockResolvedValue([
+//         Object.assign({ id: 'd376de0577681ca93614' }, profile),
+//       ]);
+//       const res = await request(server).post('/profile').send(profile);
+//       expect(res.status).toBe(200);
+//       expect(res.body.profile.id).toBe('d376de0577681ca93614');
+//       expect(Profiles.create.mock.calls.length).toBe(1);
+//     });
+//   });
+//   describe('PUT /profile', () => {
+//     it('should return 200 when profile is created', async () => {
+//       const profile = {
+//         id: 'd376de0577681ca93614',
+//         name: 'Louie Smith',
+//         email: 'louie@example.com',
+//         avatarUrl:
+//           'https://s3.amazonaws.com/uifaces/faces/twitter/hermanobrother/128.jpg',
+//       };
+//       Profiles.findById.mockResolvedValue(profile);
+//       Profiles.update.mockResolvedValue([profile]);
+//       const res = await request(server).put('/profile/').send(profile);
+//       expect(res.status).toBe(200);
+//       expect(res.body.profile.name).toBe('Louie Smith');
+//       expect(Profiles.update.mock.calls.length).toBe(1);
+//     });
+//   });
+// });
```
Please either uncomment this code or delete it.
I will not merge Zombie code.

shighetari committed:
test: unit testing for order router end points

Istott reviewed (approved):
lgtm

Commit:
fix: start mocking data for testing

Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>

Commit:
test: finish order testing

Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>

Commit:
fix: use block scope for cases

Signed-off-by: William Sedlacek <wsedlacekc@gmail.com>
